firmware: ath10k-firmware: update to 2018-02-09

heinzek opened the pull request:
This patch updates ath10k-firmware to last commit and use the
firmware-5.bin_10.4-3.5.3-00053 firmware for the QCA9888.

Tested on Archer C59v1 and C60v2.

heinzek committed:
firmware: ath10k-firmware: update to 2018-02-09

This patch updates ath10k-firmware to last commit and use the
firmware-5.bin_10.4-3.5.3-00053 firmware for the QCA9888.

Signed-off-by: Henryk Heisig <hyniu@o2.pl>

pkgadd commented:
Hi

Could you please add the same change for QCA9984 to your pull request as well? I was just about to file a pull request for this, but would prefer not to conflict with yours.

```
From ab9165adc33f92661085d3732f2f3c9e23445f1f Mon Sep 17 00:00:00 2001
From: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Date: Sun, 18 Feb 2018 04:18:18 +0100
Subject: [PATCH] firmware: ath10k-firmware: update QCA9984 firmware to
 10.4-3.5.3-00053

This patch updates ath10k-firmware to use the
firmware-5.bin_10.4-3.5.3-00053 firmware for the QCA9984.

The update fixes "ath10k_pci 0001:01:00.0: Invalid VHT mcs 15 peer
stats" spamming the kernel ring buffer at very high frequencies, but
introduces the new "ath10k_pci 0001:01:00.0: Unknown eventid: 36925".
This new warning doesn't appear to cause problems in practice and is
only emitted relatively rarely, not causing dmesg to overflow within
minutes.

Tested on the ZyXEL NBG6817; early feedback also suggests this firmware
to work well (with the same fixes and caveats) on the Netgear r7800 as
well.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
---
 package/firmware/ath10k-firmware/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/firmware/ath10k-firmware/Makefile b/package/firmware/ath10k-firmware/Makefile
index 415931bc63..e4693cc4af 100644
--- a/package/firmware/ath10k-firmware/Makefile
+++ b/package/firmware/ath10k-firmware/Makefile
@@ -314,7 +314,7 @@ define Package/ath10k-firmware-qca9984/install
 		$(PKG_BUILD_DIR)/QCA9984/hw1.0/board-2.bin \
 		$(1)/lib/firmware/ath10k/QCA9984/hw1.0/board-2.bin
 	$(INSTALL_DATA) \
-		$(PKG_BUILD_DIR)/QCA9984/hw1.0/3.4/firmware-5.bin_10.4-3.4-00104 \
+		$(PKG_BUILD_DIR)/QCA9984/hw1.0/3.5.3/firmware-5.bin_10.4-3.5.3-00053 \
 		$(1)/lib/firmware/ath10k/QCA9984/hw1.0/firmware-5.bin
 endef
 
-- 
2.16.1


```
I'll mail you the same patch in a git am'able format.

heinzek committed:
firmware: ath10k-firmware: update QCA9984 firmware to 10.4-3.5.3-00053

This patch updates ath10k-firmware to use the
firmware-5.bin_10.4-3.5.3-00053 firmware for the QCA9984.

The update fixes "ath10k_pci 0001:01:00.0: Invalid VHT mcs 15 peer
stats" spamming the kernel ring buffer at very high frequencies, but
introduces the new "ath10k_pci 0001:01:00.0: Unknown eventid: 36925".
This new warning doesn't appear to cause problems in practice and is
only emitted relatively rarely, not causing dmesg to overflow within
minutes.

Tested on the ZyXEL NBG6817; early feedback also suggests this firmware
to work well (with the same fixes and caveats) on the Netgear r7800 as
well.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>
Signed-off-by: Henryk Heisig <hyniu@o2.pl>

heinzek commented:
Thanks @pkgadd for info. I add Your changes.

pkgadd commented:
Thanks a lot.

hauke commented:
On which devices have you tested this?
I saw that the QCA FW sometimes just crashes.

pkgadd commented:
ZyXEL NBG6817, ipq8065.

```
[   46.694866] ath10k_pci 0000:01:00.0: qca9984/qca9994 hw1.0 target 0x01000000 chip_id 0x00000000 sub 168c:cafe
[   46.694930] ath10k_pci 0000:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 1
[   46.711544] ath10k_pci 0000:01:00.0: firmware ver 10.4-3.5.3-00053 api 5 features no-p2p,mfp,peer-flow-ctrl,btcoex-param,allows-mesh-bcast,no-ps crc32 4c56a386
[   48.985364] ath10k_pci 0000:01:00.0: board_file api 2 bmi_id 0:1 crc32 5eae9e4b
[   54.878535] ath10k_pci 0000:01:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
```
```

[   55.393878] ath10k_pci 0001:01:00.0: qca9984/qca9994 hw1.0 target 0x01000000 chip_id 0x00000000 sub 168c:cafe
[   55.401654] ath10k_pci 0001:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 1
[   55.415137] ath10k_pci 0001:01:00.0: firmware ver 10.4-3.5.3-00053 api 5 features no-p2p,mfp,peer-flow-ctrl,btcoex-param,allows-mesh-bcast,no-ps crc32 4c56a386
[   57.689907] ath10k_pci 0001:01:00.0: board_file api 2 bmi_id 0:2 crc32 5eae9e4b
[   63.580167] ath10k_pci 0001:01:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
```
The problem with the current firmware, QCA9984/hw1.0/3.4/firmware-5.bin_10.4-3.4-00104, is that it results in ath10k spamming dmesg with "ath10k_pci 0001:01:00.0: Invalid VHT mcs 15 peer stats" at very high frequency, overflowing the ring buffer after 15-30 minutes and leading to many supressed printk's. QCA9984/hw1.0/3.5.3/firmware-5.bin_10.4-3.5.3-00053 fixes that problem completely, but it occassionally emits "ath10k_pci 0001:01:00.0: Unknown eventid: 36925" instead. In my tests "Unknown eventid: 36925" doesn't appear to cause any immediate problems and contrary to "Invalid VHT mcs 15 peer stats" it gets only triggered relatively rarely (perhaps <5 times per day, some days not at all).

I've been running QCA9984/hw1.0/3.4/firmware-5.bin_10.4-3.4-00104 on my nbg6817 for 2 weeks by now, wireless is stable and dmesg rather clean.

heinzek commented:
@hauke i tested on Archer C59v1 and this firmware is very stable. 

```
[    9.431654] PCI: Enabling device 0000:00:00.0 (0000 -> 0002)
[    9.437817] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[    9.634176] ath10k_pci 0000:00:00.0: Direct firmware load for ath10k/QCA9888/hw2.0/firmware-6.bin failed with error -2
[    9.645269] ath10k_pci 0000:00:00.0: Falling back to user helper
[    9.712883] firmware ath10k!QCA9888!hw2.0!firmware-6.bin: firmware_loading_store: map pages failed
[   10.306158] ath10k_pci 0000:00:00.0: qca9888 hw2.0 target 0x01000000 chip_id 0x00000000 sub 0000:0000
[   10.315721] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 1
[   10.336295] ath10k_pci 0000:00:00.0: firmware ver 10.4-3.5.3-00053 api 5 features no-p2p,mfp,peer-flow-ctrl,allows-mesh-bcast,no-ps crc32 a9603656
[   10.650310] ath10k_pci 0000:00:00.0: failed to fetch board data for bus=pci,bmi-chip-id=0,bmi-board-id=20 from ath10k/QCA9888/hw2.0/board-2.bin
[   10.664304] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id 0:20 crc32 a65aa05b
[   12.815469] ath10k_pci 0000:00:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
```

hmh commented:
I have tested the FW 10.4-3.5.3-00053 on a TP-Link Archer C60v2 (BR), and this firmware is very stable in that hardware. I have not observed any of the "unknown eventid" messages in the C60v2, though.

The TP-Link Archer C60v2 is supposed to have a QCA9882 if wikidev is to be trusted, but the Archer C60v2 (BR) it is loading the QCA9888 firmware, here (logs below).

Also, the previous version of the firmware in the package would instantly crash when a station tried to register (it wouldn't even spam the kernel log: it would just crash). And it would not recover afterwards even after a reset attempt, so ath10k would give up and disable the device.

```
[   10.061756] PCI: Enabling device 0000:00:00.0 (0000 -> 0002)
[   10.067954] ath10k_pci 0000:00:00.0: pci irq legacy oper_irq_mode 1 irq_mode 0 reset_mode 0
[   10.267592] ath10k_pci 0000:00:00.0: Direct firmware load for ath10k/QCA9888/hw2.0/firmware-6.bin failed with error -2
[   10.278689] ath10k_pci 0000:00:00.0: Falling back to user helper
[   10.398889] firmware ath10k!QCA9888!hw2.0!firmware-6.bin: firmware_loading_store: map pages failed
[   10.995988] ath10k_pci 0000:00:00.0: qca9888 hw2.0 target 0x01000000 chip_id 0x00000000 sub 0000:0000
[   11.005547] ath10k_pci 0000:00:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 1
[   11.026137] ath10k_pci 0000:00:00.0: firmware ver 10.4-3.5.3-00053 api 5 features no-p2p,mfp,peer-flow-ctrl,allows-mesh-bcast,no-ps crc32 a9603656
[   11.339992] ath10k_pci 0000:00:00.0: failed to fetch board data for bus=pci,bmi-chip-id=0,bmi-board-id=20 from ath10k/QCA9888/hw2.0/board-2.bin
[   11.354056] ath10k_pci 0000:00:00.0: board_file api 1 bmi_id 0:20 crc32 65a0b85a
[   13.505014] ath10k_pci 0000:00:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
```

hmh commented:
If the update for the QCA9884 FW is delaying the merge of this PR, please update the QCA9888 firmware only, like the first version of the proposed patch for the PR did...  the QCA9884 update can be left to a separate PR.

FWIW, upstream is aware of the issues in the QCA9884, and working on them according to reports in the LKML.

pkgadd commented:
I'm certainly fine with deferring the QCA9984 patch, if it delays merging the QCA9988 update (actually I'm also sitting on a tested QCA988x firmware update I don't want to push until this is sorted).

That said, while the current QCA9984 firmware (QCA9984/hw1.0/3.4/firmware-5.bin_10.4-3.4-00104) is functional, it does emit "Invalid VHT mcs 15 peer stats" at high frequency/ overflowing the kernel message buffer within 15-30 minutes. Regardless of "Unknown eventid: 36925", QCA9984/hw1.0/3.5.3/firmware-5.bin_10.4-3.5.3-00053 is a strict improvement over the current version in LEDE.

chunkeey commented:
There's a documented/reported issue with the 10.4-3.5.3-00053 QCA9984 firmware: https://www.mail-archive.com/ath10k@lists.infradead.org/msg07983.html . As for the "Invalid VHT mcs 15 peer stats" messages: these messages have been around for a while now, older firmwares generate them as well :tired_face: ...

hauke commented:
Thank you for the pull request it was applied to master.

pkgadd commented:
Just as a clarification, QCA has now submitted a [patch to silence](https://lkml.kernel.org/r/<1522049641-19521-1-git-send-email-murugana@codeaurora.org>) "Unknown eventid: 36925" - in short, it's 'just' a sign of the firmware supporting more features than the driver (including wireless-drivers-next).
